nerfstudio/models/splatfacto.py

```diff
@@ -770,21 +778,41 @@ def get_outputs(self, camera: Cameras) -> Dict[str, Union[torch.Tensor, List]]:
         else:
             raise ValueError("Unknown rasterize_mode: %s", self.config.rasterize_mode)
 
-        rgb, alpha = rasterize_gaussians(  # type: ignore
-            self.xys,
-            depths,
-            self.radii,
-            conics,
-            num_tiles_hit,  # type: ignore
-            rgbs,
-            opacities,
-            H,
-            W,
-            BLOCK_WIDTH,
-            background=background,
-            return_alpha=True,
-        )  # type: ignore
-        alpha = alpha[..., None]
+        if self.config.background_color == "auto":
+            rgb, alpha = rasterize_gaussians(  # type: ignore
+                self.xys,
+                depths,
+                self.radii,
+                conics,
+                num_tiles_hit,  # type: ignore
+                rgbs,
+                opacities,
+                H,
+                W,
+                BLOCK_WIDTH,
+                background=torch.zeros(3, device=self.device),
```

Comment:
Is it because the rasterize_gaussians() doesn't have grad for background parameter? If so, I think the proper fix is to add bwd grad for background input in gsplat library.

Reply:
I have implimented one on bg backporp in gsplat to handle 2d bg and it just needs way more modifications than directly alpha blended them and it changes a lots of logics. I think alpha blends here might be easier and more elegant.

Reply:
I agree that 2d bg is tricky to implement in gsplat. But for just backprop the grad for bg color, I noticed you don't really need to touch cuda implementation of gsplat. There is a thin layer of wrapper, where you can just have [one line change there](https://github.com/nerfstudio-project/gsplat/blob/main/gsplat/rasterize.py#L245): by computing the grad as 

```
(v_out_img.view(-1, 3).t() @ final_Ts.view(-1, 1)).squeeze()
```

(Above line is NOT tested though)